stable_baselines3/common/vec_env/dummy_vec_env.py

```diff
@@ -73,19 +75,20 @@ def step_wait(self) -> VecEnvStepReturn:
 
     def seed(self, seed: Optional[int] = None) -> Sequence[Union[None, int]]:
         # Avoid circular import
-        from stable_baselines3.common.utils import compat_gym_seed
-
+        # from stable_baselines3.common.utils import compat_gym_seed
         if seed is None:
-            seed = np.random.randint(0, 2**32 - 1)
-        seeds = []
-        for idx, env in enumerate(self.envs):
-            seeds.append(compat_gym_seed(env, seed=seed + idx))  # type: ignore[func-returns-value]
-        return seeds
+            self.seed = [None for _ in range(len(self.envs))]  # seed to be used in the next env.reset()
```

Comment:
also the overall behavior will be the same, we still need to document the difference in the `VecEnv` page and update the changelog to say that the seed will be effective at the next reset.